openmetadata-ui/src/main/resources/ui/playwright/e2e/Features/Permissions/EntityPermissions.spec.ts

```diff
@@ -0,0 +1,160 @@
+    test.beforeAll('Setup entity', async ({ browser }) => {
+      const { apiContext, afterAction } = await performAdminLogin(browser);
+      await EntityDataClass.preRequisitesForTests(apiContext);
+      await entity.create(apiContext);
+      await afterAction();
+    });
+
+    // Allow permissions tests
+    test.describe('Allow permissions', () => {
+      test.beforeAll('Initialize allow permissions', async ({ browser }) => {
+        const page = await browser.newPage();
+        await adminUser.login(page);
+        await initializePermissions(page, 'allow');
+        await assignRoleToUser(page, testUser);
+        await page.close();
+      });
+
+      test(`${entityType} allow common operations permissions`, async ({
+        testUserPage,
+      }) => {
+        test.slow(true);
+
+        await runCommonPermissionTests(testUserPage, entity, 'allow');
+      });
+
+      // Entity-specific tests
+      if ('specificTest' in config && config.specificTest) {
+        test(`${entityType} allow entity-specific permission operations`, async ({
+          testUserPage,
+        }) => {
+          test.slow(true);
+
+          await runEntitySpecificPermissionTests(
+            testUserPage,
+            entity,
+            'allow',
+            config.specificTest as (
+              page: Page,
+              entity: EntityClass,
+              effect: 'allow' | 'deny'
+            ) => Promise<void>
+          );
+        });
+      }
+    });
+
+    // Deny permissions tests
+    test.describe('Deny permissions', () => {
+      test.beforeAll('Initialize deny permissions', async ({ browser }) => {
+        const page = await browser.newPage();
+        await adminUser.login(page);
+        await initializePermissions(page, 'deny');
+        await assignRoleToUser(page, testUser);
+        await page.close();
+      });
+
+      test(`${entityType} deny common operations permissions`, async ({
+        testUserPage,
+      }) => {
+        test.slow(true);
+
+        await runCommonPermissionTests(testUserPage, entity, 'deny');
+      });
+
+      // Entity-specific tests
+      if ('specificTest' in config && config.specificTest) {
+        test(`${entityType} deny entity-specific permission operations`, async ({
+          testUserPage,
+        }) => {
+          test.slow(true);
+
+          await runEntitySpecificPermissionTests(
+            testUserPage,
+            entity,
+            'deny',
+            config.specificTest as (
+              page: Page,
+              entity: EntityClass,
+              effect: 'allow' | 'deny'
+            ) => Promise<void>
+          );
+        });
+      }
+    });
+
+    test.afterAll('Cleanup entity', async ({ browser }) => {
+      const { apiContext, afterAction } = await performAdminLogin(browser);
+      await entity.delete(apiContext);
+      await EntityDataClass.postRequisitesForTests(apiContext);
+      await afterAction();
+    });
```

Comment:
Since we have decided to eliminate the cleanups from the tests, let's remove the afterAll blocks for both specs.

Reply:
@shrushti2000 can't we avoid conflicts with other tests? If it's affecting other tests it can also affect tests that are running concurrently with the permissions tests.

Reply:
Have removed  users and entities  cleanups from afterAll. But still kept the  roles and permissions cleanups function in afterAll, since it was interfering with `AddRoleAndAssignToUser` test.